lua/lean/commands.lua

```diff
@@ -0,0 +1,88 @@
+local components = require'lean.infoview.components'
+local infoview = require'lean.infoview'
+local leanlsp = require'lean.lsp'
+local a = require'plenary.async'
+local html = require'lean.html'
+local rpc = require'lean.rpc'
+
+local commands = {}
+
+local plain_goal = a.wrap(leanlsp.plain_goal, 3)
+local plain_term_goal = a.wrap(leanlsp.plain_term_goal, 3)
+
+-- Keep reference around to prevent garbage collection
+local bufdiv
```

Comment:
Could we instead use a mapping from popup window handle to bufdiv that is `nil`ed on `WinClosed`? You can use `vim.fn.expand('<afile>')` to get the window handle in the autocmd.

Reply:
Oh just saw your comment above, `WinClosed` is not working for the popup window?

Reply:
Hmm so it triggers when I `:q` from within the window but not when it closes automatically after I move the cursor...

Reply:
I tried `autocmd WinClosed * echo "wont be printed"`, and it does not seem to be working for the floating popups

Reply:
Could we perhaps work around this at the moment by checking `vim.api.nvim_win_is_valid()` for each handle in this table and `nil`ing accordingly whenever a new popup is created?

Reply:
If we go that route, then I'd rather check for `vim.api.nvim_buf_is_valid` so that we can make `_by_buf` non-weak.

Reply:
I've implemented that now.

Reply:
Great, that seems like a good solution for now.